src/saturn_engine/worker/topics/rabbitmq.py

```diff
@@ -189,8 +192,19 @@ async def backoff_sleep(self, attempt: int) -> bool:
     async def message_context(
         self, message: aio_pika.abc.AbstractIncomingMessage
     ) -> AsyncIterator[TopicMessage]:
-        async with message.process():
-            yield self._deserialize(message)
+        requeue: bool = False
+        if message.message_id and self.options.max_retry:
+            requeue = (
+                self.attempt_by_message[message.message_id] <= self.options.max_retry
```

Comment:
`<` no?

If `max_retry` is set to `1`, I would expect to only be requeued once (which means requeue only when `attempt_by_message` is `0`.

Reply:
Yeah but attempt_by_message will be at zero at first, then 1. So we want to retry at 1 no ?

Reply:
No, 

```
attempt = 0
requeue = attempt <= self.options.max_retry := 0 <= 1 := true
run_message()
# fail, attempt++ and requeue
```
then
```
attempt = 1
requeue = attempt <= self.options.max_retry := 1 <= 1 := true
run_message()
# fail, attempt++ and requeue
```

See how the second run we retry and will run a third time even tho max_retry is `1`

Reply:
Nah nvm you are right .. I was wrong